lib/linux_admin/network_interface.rb

```diff
@@ -0,0 +1,138 @@
+    attr_reader :interface
+
+    # @param interface [String] Name of the network interface to manage
+    def initialize(interface)
+      @interface = interface
+    end
+
+    # Retrieve the IPv4 address assigned to the interface
+    #
+    # @return [String] IPv4 address for the managed interface
+    def address
+      return nil unless (ip_output = ip_show)
+
+      cidr_ip = parse_ip_output(ip_output, /inet/, 1)
+      cidr_ip.split('/')[0] if cidr_ip
+    end
+
+    # Retrieve the IPv6 address assigned to the interface
+    #
+    # @return [String] IPv6 address for the managed interface
+    def address6(scope = :global)
+      return nil unless (ip_output = ip_show)
+
+      ip_regex = /inet6 .* scope #{scope}/
+      cidr_ip = parse_ip_output(ip_output, ip_regex, 1)
+      cidr_ip.split('/')[0] if cidr_ip
+    end
+
+    # Retrieve the MAC address associated with the interface
+    #
+    # @return [String] the MAC address
+    def mac_address
+      return nil unless (ip_output = ip_show)
+      parse_ip_output(ip_output, %r{link/ether}, 1)
+    end
+
+    # Retrieve the IPv4 sub-net mask assigned to the interface
+    #
+    # @return [String] IPv4 netmask
+    def netmask
+      return nil unless (ip_output = ip_show)
+
+      cidr_ip = parse_ip_output(ip_output, /inet/, 1)
+      IPAddr.new('255.255.255.255').mask(cidr_ip.split('/')[1]).to_s if cidr_ip
+    end
+
+    # Retrieve the IPv6 sub-net mask assigned to the interface
+    #
+    # @return [String] IPv6 netmask
+    def netmask6(scope = :global)
+      return nil unless (ip_output = ip_show)
+
+      ip_regex = /inet6 .* scope #{scope}/
+      cidr_ip = parse_ip_output(ip_output, ip_regex, 1)
+      IPAddr.new('ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff').mask(cidr_ip.split('/')[1]).to_s if cidr_ip
+    end
+
+    # Retrieve the IPv4 default gateway associated with the interface
+    #
+    # @return [String] IPv4 gateway address
+    def gateway
+      result = run(cmd("ip"), :params => ["route"])
+      return nil if result.failure?
+
+      parse_ip_output(result.output, /^default/, 2)
+    end
+
+    private
+
+    # Detects the local linux distro and returns the associated NetworkInterface subclass
+    #
+    # @return [Class] The NetworkInterface subclass associated with the local distro
+    def self.dist_class_uncached
+      if [Distros.rhel, Distros.fedora].include?(Distros.local)
+        return NetworkInterfaceRH
+      else
+        return NetworkInterfaceGeneric
+      end
+      raise "No matching class for distro #{Distros.local}"
+    end
+    private_class_method :dist_class_uncached
+
+    # Parses the output of `ip addr show`
+    #
+    # @param output [String] The command output
+    # @param regex  [Regexp] Regular expression to match the desired output line
+    # @param col    [FixNum] The whitespace delimited column to be returned
+    # @return [String] The parsed data
+    def parse_ip_output(output, regex, col)
```

Comment:
It doesn't show up in the generated html docs, but I added it for the developers.